# Objective of the analysis

We will analyze the influence of students’ geographic region on their ENEM performance.

# Data preparation

In [2]:
import pandas as pd

colunas = [
    "SG_UF_ESC",
    "NU_NOTA_CN",
    "NU_NOTA_CH",
    "NU_NOTA_LC",
    "NU_NOTA_MT",
    "NU_NOTA_REDACAO"
]
df = pd.read_csv(
    "../DADOS/enem_amostra.csv",
    sep=",",
    encoding="latin1",
    usecols=colunas)

print(df.head(5))

  SG_UF_ESC  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO
0       NaN       471.6       514.0       587.0       470.0            960.0
1       NaN       389.7       388.3       488.9       562.3            600.0
2        CE         NaN         NaN         NaN         NaN              NaN
3       NaN       448.0       527.8       570.3       511.9            320.0
4       NaN         NaN         NaN         NaN         NaN              NaN


First, we will remove the rows that contain only NaN values.

In [3]:
df = df.dropna(how="all")

Now we will create a new column called "REGION" based on the "SG_UF_ESC" column. To do this, we will use pandas’ map function to map each state to its corresponding region.

In [4]:
regioes = {
    # Norte
    "AC": "Norte",
    "AP": "Norte",
    "AM": "Norte",
    "PA": "Norte",
    "RO": "Norte",
    "RR": "Norte",
    "TO": "Norte",

    # Nordeste
    "AL": "Nordeste",
    "BA": "Nordeste",
    "CE": "Nordeste",
    "MA": "Nordeste",
    "PB": "Nordeste",
    "PE": "Nordeste",
    "PI": "Nordeste",
    "RN": "Nordeste",
    "SE": "Nordeste",

    # Centro-Oeste
    "DF": "Centro-Oeste",
    "GO": "Centro-Oeste",
    "MT": "Centro-Oeste",
    "MS": "Centro-Oeste",

    # Sudeste
    "ES": "Sudeste",
    "MG": "Sudeste",
    "RJ": "Sudeste",
    "SP": "Sudeste",

    # Sul
    "PR": "Sul",
    "RS": "Sul",
    "SC": "Sul"
}

df['REGIÃO'] = df['SG_UF_ESC'].map(regioes)
print(df.head(5))

  SG_UF_ESC  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO  \
0       NaN       471.6       514.0       587.0       470.0            960.0   
1       NaN       389.7       388.3       488.9       562.3            600.0   
2        CE         NaN         NaN         NaN         NaN              NaN   
3       NaN       448.0       527.8       570.3       511.9            320.0   
5       NaN       539.7       690.2       612.9       716.2            680.0   

     REGIÃO  
0       NaN  
1       NaN  
2  Nordeste  
3       NaN  
5       NaN  


Rows where the REGION value is NaN are not useful for our analysis, so we will remove them.

In [5]:
df = df.dropna(subset=["REGIÃO"])
print(df.head(5))   

   SG_UF_ESC  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO  \
2         CE         NaN         NaN         NaN         NaN              NaN   
19        SP       559.6       613.7       584.2       626.1            840.0   
21        CE       391.4       532.7       507.9       517.5            480.0   
26        SC         NaN         NaN         NaN         NaN              NaN   
30        RO       668.9       606.0       603.9       848.3            700.0   

      REGIÃO  
2   Nordeste  
19   Sudeste  
21  Nordeste  
26       Sul  
30     Norte  


Creating the column for the average scores.

In [6]:
df['nota_media']= df[["NU_NOTA_CN", "NU_NOTA_CH", "NU_NOTA_LC", "NU_NOTA_MT", "NU_NOTA_REDACAO"]].mean(axis=1)
print(df.head(5))

   SG_UF_ESC  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  NU_NOTA_REDACAO  \
2         CE         NaN         NaN         NaN         NaN              NaN   
19        SP       559.6       613.7       584.2       626.1            840.0   
21        CE       391.4       532.7       507.9       517.5            480.0   
26        SC         NaN         NaN         NaN         NaN              NaN   
30        RO       668.9       606.0       603.9       848.3            700.0   

      REGIÃO  nota_media  
2   Nordeste         NaN  
19   Sudeste      644.72  
21  Nordeste      485.90  
26       Sul         NaN  
30     Norte      685.42  


Now we group by region and calculate the mean of the “average_score” column.

In [7]:
media_regiao=df.groupby('REGIÃO')['nota_media'].mean().round(2)
print(media_regiao)

REGIÃO
Centro-Oeste    552.93
Nordeste        531.71
Norte           522.17
Sudeste         576.20
Sul             569.86
Name: nota_media, dtype: float64


It is observed that the Southeast and South regions present the highest average ENEM performance, while the North and Northeast show lower averages. These results suggest the existence of regional inequalities associated with socioeconomic and structural factors in Brazilian education.


# Statistical Analysis
Let’s analyze some statistics for each region.

In [8]:
df = df[df["nota_media"] > 0]
estatisticas = df.groupby("REGIÃO")["nota_media"].agg(
    media="mean",
    mediana="median",
    desvio_padrao="std",
    minimo="min",
    maximo="max",
    quantidade="count"
).round(2).sort_values(by="media", ascending=False)

estatisticas

,media,mediana,desvio_padrao,minimo,maximo,quantidade
REGIÃO,,,,,,
Sudeste,576.29,574.76,91.99,241.07,837.22,6037
Sul,569.86,564.79,88.22,297.40,812.92,1562
Centro-Oeste,553.30,547.18,94.53,243.37,844.30,1502
Nordeste,531.71,521.49,98.94,153.33,824.24,5988
Norte,522.17,514.22,89.17,182.36,785.00,1296


# Conclusion
The statistical analysis of ENEM average scores by region highlights meaningful differences in the performance of Brazilian participants.

The Southeast region showed the highest average performance (576.29) as well as the highest median (574.76), indicating a high and relatively consistent level of achievement among its students. It is followed by the South (569.86) and Central-West (553.30) regions.

The lowest averages were observed in the Northeast (531.71) and North (522.17), suggesting regional disparities in educational performance.

Standard deviations range from approximately 88 to 91 points, indicating significant score dispersion across all regions. The Northeast presented the highest standard deviation (98.94), suggesting greater heterogeneity in student performance, while the South showed the lowest (88.22), indicating more homogeneous scores.

It is also observed that, in all regions, medians are slightly lower than means. This suggests the presence of high-scoring participants influencing regional averages.

The Southeast (6037) and Northeast (5988) regions concentrate the largest number of participants in the analyzed sample.

Overall, the results suggest that socioeconomic factors, educational infrastructure, and regional inequalities may significantly impact students’ performance in the ENEM.